node/src/components/chain_synchronizer.rs

```diff
@@ -120,9 +121,13 @@ impl ChainSynchronizer<ParticipatingEvent> {
             _phantom: PhantomData,
         };
 
-        // If we're not configured to sync-to-genesis, return without doing anything.
+        // If we're not configured to sync-to-genesis, return without doing anything but announcing
+        // that the sync process has finished.
         if !synchronizer.config.sync_to_genesis() {
-            return Ok((synchronizer, Effects::new()));
+            return Ok((
```

Comment:
~~If this is called in the joiner with sync to genesis disabled, won't this announce the sync completed immediately, even though we still need to fast sync?~~ nvm, I misssed the specialization.